scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/BucketMetadata.java

```diff
@@ -206,8 +211,6 @@ public HashFunction create() {
 
   boolean isCompatibleWith(BucketMetadata other) {
     return other != null
-        // version 1 is backwards compatible with version 0
-        && (this.version <= 1 && other.version <= 1)
```

Comment:
Shouldn't this be rolled back to original 0.11 code instead of being deleted ?
```scala
&& this.version == other.version
```

Reply:
My thought process was that if a producer has already migrated to 0.12.1 and produced a few days of data, then upgrades to 0.12.2, they’ll have a mix of version: 1 and version: 0 in their metadata. If we make the version check strict again the consumers will upgrade to 0.12.2 and break on version: 1 data :(

maybe i can keep the check the way it was (0 and 1 are compatible) and just make a note that the next version bump should be to 2.

Reply:
yes we should re-introduce some checks otherwise the release will accept any version

Reply:
Sounds good - updated